### PR TITLE
Edinburgh #1319 back-port 

### PR DESCRIPTION
**Pull request by anonymouse64**
This is a back-port of #1319 on top of Edinburgh since we missed the branch cut.

In addition to the changes from #1319 it also:
* Updates all the `source-branch` spec for parts to use `edinburgh` instead of master
* Sets the `grade` to `stable` so we can promote any releases to the `edinburgh/edge` channel to candidate or stable when we are ready

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -552,6 +656,7 @@ parts:
   support-rulesengine:
     source: https://github.com/edgexfoundry/support-rulesengine.git
     source-branch: "master"
```
Comment (tonyespy) [inline snap/snapcraft.yaml]: Shouldn't this be updated to "edinburgh"?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -897,75 +1004,46 @@ parts:
         $SNAPCRAFT_PART_INSTALL/config/security-api-gateway/res/configuration.toml
 
   # DEVICE SERVICES parts
-  device-modbus:
-    source: https://github.com/edgexfoundry/device-modbus-go.git
+  device-virtual-go:
+    source: https://github.com/edgexfoundry/device-virtual-go.git
     source-depth: 1
     source-branch: master
```
Comment (tonyespy) [inline snap/snapcraft.yaml]: edinburgh?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -897,75 +1004,46 @@ parts:
...
-         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-mqtt/Attribution.txt"
+         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-virtual/Attribution.txt"
       install -DT "./LICENSE" \
-         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-mqtt/LICENSE"
+         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-virtual/LICENSE"
 
   device-random:
     source: https://github.com/edgexfoundry/device-random.git
     source-depth: 1
     source-branch: master
```
Comment (tonyespy) [inline snap/snapcraft.yaml]: edinburgh?

**Comment by anonymouse64** [comment]
I will push a commit changing all the branches to edinburgh after the edinburgh branches have been created for security-api-gateway and security-secret-store

File: snap/snapcraft.yaml (hunk 2 above)
Comment (anonymouse64) [reply snap/snapcraft.yaml]: Changed

File: snap/snapcraft.yaml (hunk 1 above)
Comment (anonymouse64) [reply snap/snapcraft.yaml]: Changed

**Comment by anonymouse64** [comment]
This PR is now ready for review. I've updated the description

**Pull request merged**